insecure/corruptible/factory_test.go

```diff
@@ -0,0 +1,217 @@
+}
+
+// TestFactoryHandleIncomingEvent_MulticastOverNetwork evaluates that the incoming multicast events to the conduit factory are routed to the
+// network adapter when no attacker registered to the factory.
+func TestFactoryHandleIncomingEvent_MulticastOverNetwork(t *testing.T) {
+	codec := cbor.NewCodec()
+	// corruptible conduit factory with no attacker registered.
+	f := NewCorruptibleConduitFactory(unittest.Logger(), flow.BftTestnet, unittest.IdentifierFixture(), codec)
+
+	adapter := &mocknetwork.Adapter{}
+	err := f.RegisterAdapter(adapter)
+	require.NoError(t, err)
+
+	event := &message.TestMessage{Text: "this is a test message"}
+	channel := network.Channel("test-channel")
+	targetIds := unittest.IdentifierListFixture(10)
+
+	params := []interface{}{channel, event, uint(3)}
+	for _, id := range targetIds {
+		params = append(params, id)
+	}
+
+	adapter.On("MulticastOnChannel", params...).Return(nil).Once()
+
+	err = f.HandleIncomingEvent(event, channel, insecure.Protocol_MULTICAST, uint32(3), targetIds...)
+	require.NoError(t, err)
+
+	testifymock.AssertExpectationsForObjects(t, adapter)
+}
+
+// TestProcessAttackerMessage evaluates that conduit factory relays the messages coming from the attacker to its underlying network adapter.
+func TestProcessAttackerMessage(t *testing.T) {
+	codec := cbor.NewCodec()
+	// corruptible conduit factory with no attacker registered.
+	f := NewCorruptibleConduitFactory(unittest.Logger(), flow.BftTestnet, unittest.IdentifierFixture(), codec)
+
+	adapter := &mocknetwork.Adapter{}
+	err := f.RegisterAdapter(adapter)
+	require.NoError(t, err)
+
+	event := &message.TestMessage{Text: "this is a test message"}
+	channel := network.Channel("test-channel")
+	targetIds := unittest.IdentifierListFixture(10)
+
+	params := []interface{}{channel, event, uint(3)}
+	for _, id := range targetIds {
+		params = append(params, id)
+	}
+
+	adapter.On("MulticastOnChannel", params...).Return(nil).Once()
+
+	// imitates an RPC call from the attacker
+	msg, err := f.eventToMessage(event, channel, insecure.Protocol_MULTICAST, uint32(3), targetIds...)
+	require.NoError(t, err)
+
+	err = f.processAttackerMessage(msg)
+	require.NoError(t, err)
+
+	testifymock.AssertExpectationsForObjects(t, adapter)
+}
+
+// TestEngineClosingChannel evaluates that factory closes the channel whenever the corresponding engine of that channel attempts
+// on closing it.
+func TestEngineClosingChannel(t *testing.T) {
+	codec := cbor.NewCodec()
+	// corruptible conduit factory with no attacker registered.
+	f := NewCorruptibleConduitFactory(unittest.Logger(), flow.BftTestnet, unittest.IdentifierFixture(), codec)
+
+	adapter := &mocknetwork.Adapter{}
+	err := f.RegisterAdapter(adapter)
+	require.NoError(t, err)
+
+	channel := network.Channel("test-channel")
+
+	adapter.On("UnRegisterChannel", channel).Return(nil).Once()
+
+	err = f.EngineClosingChannel(channel)
+	require.NoError(t, err)
+
+	testifymock.AssertExpectationsForObjects(t, adapter)
```

Comment:
Can you add a comment here to explain what we're asserting?

Reply:
added here: https://github.com/onflow/flow-go/pull/2033/commits/f3128ce727f942ddbe9726023f661c89e2054ed1